object_storage_api/services/image.py

```diff
@@ -75,6 +75,25 @@ def create(self, image_metadata: ImagePostMetadataSchema, upload_file: UploadFil
 
         return ImageSchema(**image_out.model_dump())
 
+    def get(self, image_id: str) -> ImageGetUrlInfoSchema:
+        """
+        Retrieve an image with its presigned url by its ID.
+        :param: ID of the image to retrieve.
+        :return: An image or None if no image is retrieved.
+        """
+        message = "Image not found"
+        try:
+            image = self._image_repository.get(image_id=image_id)
+            if not image:
+                raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message)
+            presigned_url = self._image_store.create_presigned_url(
+                object_key=image.object_key, file_name=image.file_name
+            )
+            return ImageGetUrlInfoSchema(**image.model_dump(), url=presigned_url)
+        except InvalidObjectIdError as exc:
+            logger.exception("The ID is not a valid ObjectId value")
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message) from exc
```

Comment:
I would create a new exception for this rather than raising a HTTPException in the service as these should be reserved to the router as services could in theory be run independently of the actual API. If we followed ims-api then this wouldn't error here and the response would be handled in the router. Otherwise `MissingRecordError` from IMS would work but perhaps that would make more sense if it was raised inside the repo layer. So maybe `MissingEntityError` if we keep it in the service.